librb/include/rb_lib.h

```diff
@@ -228,6 +228,15 @@ while(0)
 #define UINT32_MAX (4294967295U)
 #endif
 
+/* For those unfamiliar with GNU format attributes, a is the 1 based
+ * argument number of the format string, and b is the 1 based argument
+ * number of the variadic ... */
+#ifdef __GNUC__
+#define AFP(a,b) __attribute__((format (printf, a, b)))
```

Comment:
Sure but that just adds to this diff because the calls are all over... There will have to be a complete macro/define refactor next.